pkg/configurator/client.go

```diff
@@ -21,44 +17,16 @@ import (
 )
 
 // NewConfigurator implements configurator.Configurator and creates the Kubernetes client to manage namespaces.
-func NewConfigurator(configClient configClientset.Interface, stop <-chan struct{}, osmNamespace, meshConfigName string,
+func NewConfigurator(informerCollection *informers.InformerCollection, osmNamespace, meshConfigName string,
```

Comment:
I actually think we should keep them separate since the unexported function returns a concrete type instead of an interface, allowing tests to call unexported methods on said concrete type

Reply:
from Rob Pike "accept interfaces return structs", let's just return the struct, there's no need to provide additional restrictions on top of us choosing which methods to export

Reply:
Fair enough!